### PR TITLE
decorate - compose decorators for a single prop

### PR DESCRIPTION
**Pull request by ramybenaroya**
PR checklist:

* [x] Added unit tests
* [ ] Updated changelog
* [x] Updated docs (either in the description of this PR as markdown, or as separate PR on the `gh-pages` branch. Please refer to this PR). For new functionality, at least [API.md](https://github.com/mobxjs/mobx/blob/gh-pages/docs/refguide/api.md) should be updated
* [x] Added typescript typings
* [x] Verified that there is no significant performance drop (`npm run perf`)

## Changes
- Reduce multiple decorators to a single decorator
- Added a test for multiple decorators (@action + custom) on a function property
- Added a test for multiple decorators (@observable + @serializable) on
  a regular property
- Added a usage example (+ caveat warning) in readme under `decorate` section

Usage Example: 
```javascript
import { decorate, observable } from "mobx"
import { serializable, primitive } from "serializr"
import persist from 'mobx-persist';

class Todo {
    id = Math.random();
    title = "";
    finished = false;
}
decorate(Todo, {
    title: [serializable(primitive), persist('object'), observable],
    finished: observable
})
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -67,6 +67,7 @@
     "rollup": "^0.41.6",
     "rollup-plugin-filesize": "^1.3.2",
     "rollup-plugin-node-resolve": "^3.0.0",
+    "serializr": "^1.3.0",
```
Comment (ramybenaroya) [inline package.json]: I added `serializr` as a dev dependency just for this [test](https://github.com/mobxjs/mobx/pull/1652/files#diff-2492cdcb925c1d209eec08acc8ff1320R428), so I can compose `@observable` with another decorator `@serializable(primitive())`.

File: README.md [hunk 2]
```diff
@@ -107,6 +107,24 @@ decorate(Todo, {
...
+    id = Math.random();
+    title = "";
+    finished = false;
+}
+decorate(Todo, {
+    title: [serializable(primitive), persist('object'), observable],
+    finished: [serializable(primitive), observable]
+})
+```
+Note: Composing decorators can sometimes lead to strange things when one decorator modifies the [property descriptor](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty) in a way that is unexpected for the other decorators.
```
Comment (ramybenaroya) [inline README.md]: Not so sure that my explanation is clear enough. Would love to get a feedback about it.

File: package.json (hunk 1 above)
Comment (mweststrate) [reply package.json]: N.P.

File: README.md (hunk 2 above)
Comment (mweststrate) [reply README.md]: Note: not all decorators can be composed together, and this functionality is just best-effort. Some decorators affect the instance directly and can 'hide' the effect of other decorators that only change the prototype

File: README.md [hunk 3]
```diff
@@ -107,6 +107,24 @@ decorate(Todo, {
 })
 ```
 
+For applying multiple decorators on a single property, you can pass an array of decorators.
```
Comment (mweststrate) [inline README.md]: Please mension what the application order is. I assume serializable is applied first, then persist and finally observable?

**Comment by mweststrate** [comment]
Some small comments, further looking great! 
Note that if you need to support this feature for MobX, a second PR targeting `mobx4-master` should be opened.

Thanks!

File: README.md (hunk 3 above)
Comment (ramybenaroya) [reply README.md]: correct

**Comment by ramybenaroya** [comment]
Thanks for the feedback @mweststrate.
> if you need to support this feature for MobX...     

I didn't get what you meant by that. Can you elaborate about "support for MobX"?

**Comment by mweststrate** [comment]
@ramybenaroya sorry, meant MobX _4_. In other words; master represents 5, but 4 is still maintained as well so it might be interesting to backport this one. (if this branch is merged with squash, it can probably just be cherry-picked)

**Pull request merged**